modules/apps/headless/headless-admin-list-type/headless-admin-list-type-impl/src/main/java/com/liferay/headless/admin/list/type/internal/resource/v1_0/ListTypeDefinitionResourceImpl.java

```diff
@@ -169,13 +169,17 @@ public ListTypeDefinition putListTypeDefinitionByExternalReferenceCode(
 		com.liferay.list.type.model.ListTypeDefinition
 			serviceBuilderListTypeDefinition =
 				_listTypeDefinitionService.
-					getListTypeDefinitionByExternalReferenceCode(
+					fetchListTypeDefinitionByExternalReferenceCode(
 						listTypeDefinitionExternalReferenceCode,
 						contextCompany.getCompanyId());
 
-		return putListTypeDefinition(
-			serviceBuilderListTypeDefinition.getListTypeDefinitionId(),
-			listTypeDefinition);
+		if (serviceBuilderListTypeDefinition != null) {
+			return putListTypeDefinition(
+				serviceBuilderListTypeDefinition.getListTypeDefinitionId(),
+				listTypeDefinition);
+		}
+
+		return postListTypeDefinition(listTypeDefinition);
```

Comment:
I think you forget to add this line:
`listTypeDefinition.setExternalReferenceCode(externalReferenceCode);`
Please, see this [example](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java#L454-L474) :+1: